prow/spyglass/podlogartifact.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spyglass
+
+import (
+	"bytes"
+	"io"
+	"net/url"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/deck/jobs"
+	"k8s.io/test-infra/prow/spyglass/viewers"
+)
+
+// PodLogArtifact holds data for reading from a specific pod log
+type PodLogArtifact struct {
+	name      string
+	buildID   string
+	podName   string
+	sizeLimit int64
+	ja        *jobs.JobAgent
+}
+
+// NewPodLogArtifact creates a new PodLogArtifact
+func NewPodLogArtifact(jobName string, buildID string, podName string, sizeLimit int64, ja *jobs.JobAgent) *PodLogArtifact {
+	if jobName == "" {
+		logrus.Error("Must specify non-empty jobName")
+		return &PodLogArtifact{}
+	}
+	if buildID == "" {
+		logrus.Error("Must specify non-empty buildID")
+		return &PodLogArtifact{}
+	}
+	return &PodLogArtifact{
+		name:      jobName,
+		buildID:   buildID,
+		podName:   podName,
+		sizeLimit: sizeLimit,
+		ja:        ja,
+	}
+}
+
+// CanonicalLink returns a link to where pod logs are streamed
+func (a *PodLogArtifact) CanonicalLink() string {
+	if a.name != "" && a.buildID != "" {
+		q := url.Values{
+			"job": []string{a.name},
+			"id":  []string{a.buildID},
+		}
+		u := url.URL{
+			Path:     "/log",
+			RawQuery: q.Encode(),
+		}
+		return u.String()
+	}
+	logrus.Error("Error getting link to pod log, empty job name or build ID")
```

Comment:
You should be able to validate the last two cases in the constructor, right?

Reply:
yes, fix incoming